Sources/Plasma/NucleusLib/pnSceneObject/plCoordinateInterface.cpp

```diff
@@ -382,9 +386,10 @@ static inline hsMatrix44 IMatrixMul34(const hsMatrix44& lhs, const hsMatrix44& r
 {
     hsMatrix44 ret;
     ret.NotIdentity();
-    ret.fMap[3][0] = ret.fMap[3][1] = ret.fMap[3][2] = 0;
-    ret.fMap[3][3] = 1.f;
-
+    
+#ifdef HS_BUILD_FOR_APPLE
+    vDSP_mmul((const float*)lhs.fMap, 1, (const float*)rhs.fMap, 1, (float*)&(ret.fMap), 1, 3, 4, 4);
```

Comment:
The invocation here is a little odd in since technically these are both 4x4 matrices.
https://developer.apple.com/documentation/accelerate/1449984-vdsp_mmul?language=objc

(See: __M, __N, __P.)

Seems to work though.

Reply:
Another copy of this function also exists in `plDrawableSpans`. I wonder if it would be worth moving `IMatrixMul34` into a header file in CoreLib and de-duplicating the implementations?

Reply:
So far the plDrawableSpans implementation doesn't seem to be showing up on CPU traces. But I think it would be good to de-duplicate them. I'll move this into hsMatrix44 in since that seems to be the type used for this algorithm (even though technically it's a 34.)